File: kurs programowania/lista3/zadanie1/Pieciokat.java
public class Pieciokat extends Figura {

    public int bok;

    public Pieciokat(int bok) {

        if (bok < 0)
            throw new IllegalArgumentException();

        this.bok = bok;
    }

    public double pole() {

        return bok * bok * Math.sqrt(25 + 10 * Math.sqrt(5)) / 4;
    }

    public double obwod() {

        return 5 * bok;
    }
}

File: kurs programowania/lista3/cpp/Pieciokat.cpp
#include "Pieciokat.hpp"

#include <iostream>
#include <math.h>

Pieciokat::Pieciokat(int a)
{
    if (a < 0)
        throw std::runtime_error("illegal_args");

    this->a = a;
}

double Pieciokat::pole()
{
    return a * a * sqrt(25 + 10 * sqrt(5)) / 4;
}

double Pieciokat::obwod()
{
    return 5 * a;
}